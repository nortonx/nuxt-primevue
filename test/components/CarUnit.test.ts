import { shallowMount, type VueWrapper } from "@vue/test-utils";
import CarUnit from "@/components/CarUnit.vue";
import { describe, it, expect, beforeEach } from "vitest";

describe("Car component", () => {
  let wrapper: VueWrapper<unknown>;

  beforeEach(() => {
    wrapper = shallowMount(CarUnit, {
      props: {
        carModel: "Toyota",
        carPlate: "ABC-1234",
        carColor: "black",
        carOwner: "John Doe",
      },
    });
  });

  it("should render correctly and match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render correctly with props and match snapshot", () => {
    expect(wrapper.find(".car-model").text()).toBe(
      `Model: ${wrapper.props().carModel}`,
    );
    expect(wrapper.find(".car-plate").text()).toBe(
      `Plate: ${wrapper.props().carPlate}`,
    );
    expect(wrapper.find(".car-color").text()).toBe(
      `Color: ${wrapper.props().carColor}`,
    );
    expect(wrapper.find(".car-owner").text()).toBe(
      `Owner: ${wrapper.props().carOwner}`,
    );

    expect(wrapper.html()).toMatchSnapshot();
  });
});
