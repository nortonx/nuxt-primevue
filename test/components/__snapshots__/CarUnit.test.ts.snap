// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Car component > should render correctly and match snapshot 1`] = `
"<div data-v-8e5aa4ad="" class="car-item border-1 border-round-sm p-2 w-2 flex flex-column justify-content-center align-items-center m-3"><img data-v-8e5aa4ad="" src="/assets/car.svg" alt="Toyota" class="car-image p-1">
  <div data-v-8e5aa4ad="" class="car-info">
    <div data-v-8e5aa4ad="" class="car-model"><span data-v-8e5aa4ad="" class="car-property font-bold">Model</span>: Toyota</div>
    <div data-v-8e5aa4ad="" class="car-plate"><span data-v-8e5aa4ad="" class="car-property font-bold">Plate</span>: <code data-v-8e5aa4ad="">ABC-1234</code></div>
    <div data-v-8e5aa4ad="" class="car-color"><span data-v-8e5aa4ad="" class="car-property font-bold">Color</span>: <span data-v-8e5aa4ad="" class="car-property-value" style="border: 2px solid black; padding-left: 2px; padding-right: 2px;">black</span></div>
    <div data-v-8e5aa4ad="" class="car-owner"><span data-v-8e5aa4ad="" class="car-property font-bold">Owner</span>: John Doe</div>
  </div>
</div>"
`;

exports[`Car component > should render correctly with props and match snapshot 1`] = `
"<div data-v-8e5aa4ad="" class="car-item border-1 border-round-sm p-2 w-2 flex flex-column justify-content-center align-items-center m-3"><img data-v-8e5aa4ad="" src="/assets/car.svg" alt="Toyota" class="car-image p-1">
  <div data-v-8e5aa4ad="" class="car-info">
    <div data-v-8e5aa4ad="" class="car-model"><span data-v-8e5aa4ad="" class="car-property font-bold">Model</span>: Toyota</div>
    <div data-v-8e5aa4ad="" class="car-plate"><span data-v-8e5aa4ad="" class="car-property font-bold">Plate</span>: <code data-v-8e5aa4ad="">ABC-1234</code></div>
    <div data-v-8e5aa4ad="" class="car-color"><span data-v-8e5aa4ad="" class="car-property font-bold">Color</span>: <span data-v-8e5aa4ad="" class="car-property-value" style="border: 2px solid black; padding-left: 2px; padding-right: 2px;">black</span></div>
    <div data-v-8e5aa4ad="" class="car-owner"><span data-v-8e5aa4ad="" class="car-property font-bold">Owner</span>: John Doe</div>
  </div>
</div>"
`;
